### PR TITLE
max_difference

### PR DESCRIPTION
**Pull request by megha070**
Solves #396

### CODE REVIEW

**Comment by sahilbansal17** [comment]
Kindly drop the commits from the previous pull request being shown here.
Read [this section on editing commits](https://github.com/susam/gitpr#edit-commits).

After doing so, you have to do [force push](https://github.com/susam/gitpr#force-push) to the topic branch.

**Comment by megha070** [comment]
I couldn't get this. Please help me with the same.

**Comment by sahilbansal17** [comment]
There are these commits for **coin_change** which are to be dropped in this new branch. 
1. So, first checkout to this branch.
`git checkout branchName`

2. Then, perform the **rebase** command:
`git rebase -i HEAD~x`
Here, **x** denotes the no. of commits you want to go back. After which, a text file will open, Just read the instructions and **drop** the unnecessary commits.

3. Finally, do force push to this branch.
`git push -f origin branchName`

**Comment by megha070** [comment]
I still couldn't figure this out. I don't know if it is okay to delete that particular file but I deleted it.

**Comment by sahilbansal17** [comment]
Did you try running those commands?

**Comment by megha070** [comment]
yes, I did 
the same message pops up every time
"everything up-to-date"

**Comment by megha070** [comment]
gitttt
I am attaching screenshot of the terminal window.

**Comment by sahilbansal17** [comment]
Instead of 
`git rebase --continue`

Use
`git rebase -i HEAD~6`

**Comment by megha070** [comment]
I did that too still the same.

**Comment by megha070** [comment]
Oh thank you, I guess it's done.

**Comment by sahilbansal17** [comment]
I am trying to fix it, there is still something going wrong.

**Comment by megha070** [comment]
Oh okayy!

**Comment by sahilbansal17** [comment]
Now, you can see there are just 2 relevant commits in this branch. 

image

File: Algorithms/Stacks/maxdiff.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     for(int i=0;i<t;i++)
+     {
+         int ch=Math.abs(right.get(i)-left.get(t-1-i));
+         //get absolute difference between right and left smaller elements
+         ans=Math.max(ans,ch);
+     }
+     return ans;
+    }
+    //function to get an arraylist consisting of left smaller elements
+  public static ArrayList<Integer> getarray(int a[],int n)
```
Comment (sahilbansal17) [inline Algorithms/Stacks/maxdiff.java]: Kindly indent the code properly. Some functions are indented by 4 spaces and others by 2 spaces. 
Make sure there is consistency.

File: Algorithms/Stacks/maxdiff.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+//This code finds the maximum absolute difference between 
+//nearest left and right smaller element of every element in an array
+import java.util.*;
+import java.io.*;
+class maxdiff
+{
+    public static void main(String args[])
+    {
+        Scanner s=new Scanner(System.in);
```
Comment (sahilbansal17) [inline Algorithms/Stacks/maxdiff.java]: Use proper spaces everywhere as mentioned in this [section](https://github.com/sahilbansal17/Competitive_Coding/wiki/Coding-Standards#spaces) in coding standards.

**Comment by sahilbansal17** [comment]
@megha070 Please make the requested changes and add one more commit. 
Do clone your repository once again locally to make sure the branch is properly updated with the forced commits.

**Comment by megha070** [comment]
@sahilbansal17 Please review this PR, I have made the necessary changes.

**Pull request merged**